### PR TITLE
Update ibc-go to v8

### PR DESCRIPTION
**Pull request by dongrie**
Made a series of modifications to support ibc-go/v8

### CODE REVIEW

File: tests/chains/tendermint/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG GOLANG_VER=1.20.5
+ARG GOLANG_VER=1.21
```
Comment (siburu) [inline tests/chains/tendermint/Dockerfile:1]: Please specify a patch version for making tests reproducible.
`1.21.8` looks good at the moment.

File: tests/chains/tendermint/simapp/export.go [hunk 2]
```diff
@@ -71,12 +74,19 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
 
 	// withdraw all validator commission
 	app.StakingKeeper.IterateValidators(ctx, func(_ int64, val stakingtypes.ValidatorI) (stop bool) {
-		_, _ = app.DistrKeeper.WithdrawValidatorCommission(ctx, val.GetOperator())
+		valBz, err := app.StakingKeeper.ValidatorAddressCodec().StringToBytes(val.GetOperator())
```
Comment (siburu) [inline tests/chains/tendermint/simapp/export.go:77]: At L77 you used `app.StakingKeeper.ValidatorAddressCodec().StringToBytes` to convert `val.GetOperator()` into `sdk.ValAddress`.
On the other hand, at L115, you used `sdk.ValAddressFromBech32` for the same purpose.

What is difference between them? Which one is the best choice? Why is it so?

File: tests/chains/tendermint/simapp/export.go [hunk 3]
```diff
@@ -158,13 +182,13 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
...
+	iter := storetypes.KVStoreReversePrefixIterator(store, stakingtypes.ValidatorsKey)
 	counter := int16(0)
 
 	for ; iter.Valid(); iter.Next() {
 		addr := sdk.ValAddress(stakingtypes.AddressFromValidatorsKey(iter.Key()))
-		validator, found := app.StakingKeeper.GetValidator(ctx, addr)
-		if !found {
+		validator, err := app.StakingKeeper.GetValidator(ctx, addr)
+		if err != nil {
 			panic("expected validator, not found")
```
Comment (siburu) [inline tests/chains/tendermint/simapp/export.go:192]: `panic(err)` is better.

File: tests/chains/tendermint/simapp/simd/cmd/root.go [hunk 4]
```diff
@@ -5,50 +5,63 @@ import (
...
+	appOpts := simtestutil.NewAppOptionsWithFlagHome(tempDir())
+	homePath, ok := appOpts.Get(flags.FlagHome).(string)
+	if !ok || homePath == "" {
+		panic("application home not set")
+	}
+	tempApp := simapp.NewSimApp(log.NewNopLogger(), dbm.NewMemDB(), nil, true, map[int64]bool{}, homePath, uint(1), appOpts)
+	interfaceRegistry := tempApp.InterfaceRegistry()
+	appCodec := codec.NewProtoCodec(interfaceRegistry)
+	txConfig := tempApp.GetTxConfig()
+	amino := codec.NewLegacyAmino()
```
Comment (siburu) [inline tests/chains/tendermint/simapp/simd/cmd/root.go:60]: These codes appear to be a bit different from root.go of ibc-go@v8.0.0 or v8.1.0. Why?

File: tests/chains/tendermint/simapp/types.go
Comment (siburu) [inline tests/chains/tendermint/simapp/types.go:1]: I think this file (types.go) is no longer necessary.

File: tests/chains/tendermint/simapp/app.go
Comment (siburu) [inline tests/chains/tendermint/simapp/app.go:1]: I have taken a diff between this file and `github.com/cosmos/ibc-go/testing/simapp/app.go` (v8.0.0) and found that too many lines differ from each other.
Please try the same between the current version (main branch) of yui-relayer and ibc-go@v7.2.0.
You will find that the difference is very small.

The simapp in yui-relayer should be based on the simapp of ibc-go (v8.0.0) and the difference between them should be minimized.

**Review by siburu — COMMENTED** [review]
@dongrie Sorry for late review. I put some comments.

File: tests/chains/tendermint/simapp/export.go (hunk 2 above)
Comment (siburu) [reply tests/chains/tendermint/simapp/export.go:77]: According to discussion, the simapp from ibc-go@v8 also uses different methods for conversion.
Therefore you need not fix this code. Thank you.

File: chains/tendermint/msg.go [hunk 5]
```diff
@@ -44,11 +48,11 @@ func (r *MsgResult) Events() []core.MsgEventLog {
...
 
-func parseMsgEventLogs(logs sdk.ABCIMessageLogs, msgIndex uint32) ([]core.MsgEventLog, error) {
+func parseMsgEventLogs(events []abcitypes.Event, msgIndex uint32) ([]core.MsgEventLog, error) {
 	var msgEventLogs []core.MsgEventLog
-	for _, log := range logs {
-		if msgIndex == log.MsgIndex {
-			for _, ev := range log.Events {
+	for _, ev := range events {
+		for _, attr := range ev.Attributes {
+			if attr.Key == MsgIndexAttributeKey && attr.Value == strconv.FormatUint(uint64(msgIndex), 10) {
```
[the remark this answers is not in the document]
Comment (dongrie) [reply chains/tendermint/msg.go]: Fixed
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/afb578e9cde1aed70e69e63c9d2ef24a13bc27aa

File: chains/tendermint/query.go [hunk 6]
```diff
@@ -121,7 +121,7 @@ func (c *Chain) QueryBalance(ctx core.QueryContext, addr sdk.AccAddress) (sdk.Co
 		Offset:     0,
 		Limit:      1000,
 		CountTotal: true,
-	})
+	}, false)
```
[the remark this answers is not in the document]
Comment (dongrie) [reply chains/tendermint/query.go]: Fixed
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/afb578e9cde1aed70e69e63c9d2ef24a13bc27aa

File: tests/chains/tendermint/Dockerfile (hunk 1 above)
Comment (dongrie) [reply tests/chains/tendermint/Dockerfile]: Use 1.21.8
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/afb578e9cde1aed70e69e63c9d2ef24a13bc27aa

File: tests/chains/tendermint/simapp/simd/cmd/root.go (hunk 4 above)
Comment (dongrie) [reply tests/chains/tendermint/simapp/simd/cmd/root.go]: Fixed
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/afb578e9cde1aed70e69e63c9d2ef24a13bc27aa

File: tests/chains/tendermint/simapp/types.go
Comment (dongrie) [reply tests/chains/tendermint/simapp/types.go:1]: Removed
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/afb578e9cde1aed70e69e63c9d2ef24a13bc27aa

File: tests/chains/tendermint/simapp/app.go
Comment (dongrie) [reply tests/chains/tendermint/simapp/app.go:1]: Fixed
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/afb578e9cde1aed70e69e63c9d2ef24a13bc27aa

File: chains/tendermint/msg.go [hunk 7]
```diff
@@ -44,23 +48,30 @@ func (r *MsgResult) Events() []core.MsgEventLog {
...
 
-func parseMsgEventLog(ev sdk.StringEvent) (core.MsgEventLog, error) {
+func msgIndexOf(event abcitypes.Event) string {
+	for _, attr := range event.Attributes {
+		if attr.Key == MsgIndexAttributeKey {
+			return attr.Value
+		}
+	}
+	return ""
+}
```
[the remark this answers is not in the document]
Comment (dongrie) [reply chains/tendermint/msg.go]: Fixed
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/68732a2a4f5aac3b987e94b1c85297463d41ed8e

File: go.mod [hunk 8]
```diff
@@ -1,80 +1,86 @@
...
+	github.com/prometheus/client_golang v1.17.0
 	github.com/spf13/cobra v1.7.0
 	github.com/spf13/viper v1.16.0
 	go.opentelemetry.io/otel v1.16.0
 	go.opentelemetry.io/otel/exporters/prometheus v0.39.0
 	go.opentelemetry.io/otel/metric v1.16.0
 	go.opentelemetry.io/otel/sdk/metric v0.39.0
-	golang.org/x/exp v0.0.0-20230515195305-f3d0a9c9a5cc
-	golang.org/x/sync v0.1.0
+	golang.org/x/exp v0.0.0-20231006140011-7918f672742d
```
Comment (siburu) [inline go.mod:27]: Can we switch from `golang.org/x/exp/slog` to `log/slog` in the standard library?

File: tests/chains/tendermint/simapp/app.go
Comment (siburu) [inline tests/chains/tendermint/simapp/app.go:1]: Could you also update `simapp/README.md` ?

**Comment by siburu** [comment]
@dongrie Could you also update simapp/README.md ?

File: go.mod (hunk 8 above)
Comment (dongrie) [reply go.mod]: Change `golang.org/x/exp/slog` to `log/slog`
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/fdaee6cb0c631ab36c7e0af7d26d3615ff849804

**Comment by dongrie** [comment]
> @dongrie Could you also update simapp/README.md ?

Updated simapp/README.md
https://github.com/hyperledger-labs/yui-relayer/pull/134/commits/1020f7f717eae5845c5cdf359fdaad620246ea9b

**Review by siburu — APPROVED** [review]
@dongrie LGTM. Thank you!

**Pull request merged**